File: runtimejs/java/lang/Enum.js
//reference// java/lang/IllegalArgumentException
//load// java/lang/Object
var java_lang_Enum = function()
{   this._name = null;
    this._ordinal = 0;
};
_defclass (java_lang_Enum, java_lang_Object, null,
{  _2: function(name,ordinal) 
    {   this._name = name;
        this._ordinal = ordinal;
        return this;
    },    

    equals_1 : function(other) 
    {   return this===other;
    },
    
    hashCode_0 : function() {
        return this._ordinal;
    },
    
    name_0: function() {
        return this._name;
    },
    
    ordinal_0: function() {
        return this._ordinal;
    },
    
    toString_0: function() {
        return this._name;
    }    
}); 

File: runtimecs/java/lang/Enum.cs
namespace java.lang 
{
    public class Enum 
    {
        private readonly System.String n;
        private readonly int o;
               
        public Enum(System.String name, int ordinal) 
        {   
            n = name;
            o = ordinal;
        }

        public override bool Equals(System.Object obj) 
        {   
            return this==obj;
        }

        public override int GetHashCode()
        {   
            return o;
        }

        public override System.String ToString()
        {   
            return n;
        }

        public System.String name()
        {   
            return n;
        }

        public int ordinal()
        {   
            return o;
        }
    }
}
